enarx-keep/shims/shim-sev/src/gdt.rs

```diff
@@ -2,46 +2,61 @@
 
 //! Global Descriptor Table init
 
-use crate::shim_stack::init_shim_stack;
+use crate::shim_stack::init_stack_with_guard;
 use crate::syscall::_syscall_enter;
 use core::ops::Deref;
-use memory::Page;
+use units::bytes;
 use x86_64::instructions::segmentation::{load_ds, load_es, load_fs, load_gs, load_ss, set_cs};
 use x86_64::instructions::tables::load_tss;
 use x86_64::registers::model_specific::{KernelGsBase, LStar, SFMask, Star};
 use x86_64::registers::rflags::RFlags;
 use x86_64::structures::gdt::{
     Descriptor, DescriptorFlags, GlobalDescriptorTable, SegmentSelector,
 };
+use x86_64::structures::paging::{Page, PageTableFlags, Size2MiB, Size4KiB};
 use x86_64::structures::tss::TaskStateSegment;
-use x86_64::{PrivilegeLevel, VirtAddr};
+use x86_64::{align_up, PrivilegeLevel, VirtAddr};
 
-const STACK_NUM_PAGES: usize = 5;
-static mut STACKS: [[Page; STACK_NUM_PAGES]; 7] = [[Page::zeroed(); STACK_NUM_PAGES]; 7];
+/// The virtual address of the main kernel stack
+pub const SHIM_STACK_START: u64 = 0xFFFF_FF48_4800_0000;
+
+/// The size of the main kernel stack
+#[allow(clippy::integer_arithmetic)]
+pub const SHIM_STACK_SIZE: u64 = bytes![4; MiB];
```

Comment:
Should we integrate this clippy allow into the bytes macro directly?

Reply:
@npmccallum That would be perfect!

Reply:
https://github.com/enarx/nbytes Patches welcome!

Reply:
I just don't know how